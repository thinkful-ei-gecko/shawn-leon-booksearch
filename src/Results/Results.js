import React, { Component } from 'react'
import Book from './Book/Book';
import './Results.css';

export default class Results extends Component {
    render() {
        return (
            <div>
                {console.log(`Results Props: ${this.props.stateData}`)}
                <ul>
                    {this.props.stateData.map((book, i) => <Book bookData={book} key={i}></Book>)}
                </ul>
            </div>
        )
    }
}

