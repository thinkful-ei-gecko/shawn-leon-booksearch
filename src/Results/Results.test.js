import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Results from './Results';

const fakeBookData = {
  bookInfo: [
    {
      title: 'Apples',
      picture: 'http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      description: 'who loves apples',
      price: '501 usd',
      author: 'apple lady1'
    },
    {
      title: 'Apples2',
      picture: 'http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      description: 'who loves apples2',
      price: '502 usd',
      author: 'apple lady2'
    },
    {
      title: 'Apples3',
      picture: 'http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      description: 'who loves apples3',
      price: '503 usd',
      author: 'apple lady3'
    },
    {
      title: 'Apples4',
      picture: 'http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      description: 'who loves apples4',
      price: '504 usd',
      author: 'apple lady4'
    },
  ]
}

describe('<Results />', () => {
  //smoke test with data?
  it('renders book list with data', () => {
    const div= document.createElement('div');
    ReactDOM.render(<Results stateData={fakeBookData.bookInfo} />,div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot with enzyme
  it('renders the Results UI with data', () => {
    const wrapper = shallow(<Results stateData={fakeBookData.bookInfo} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});