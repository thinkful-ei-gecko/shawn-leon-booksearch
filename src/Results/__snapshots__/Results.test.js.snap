// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> renders the Results UI with data 1`] = `
<div>
  <ul>
    <Book
      bookData={
        Object {
          "author": "apple lady1",
          "description": "who loves apples",
          "picture": "http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
          "price": "501 usd",
          "title": "Apples",
        }
      }
      key="0"
    />
    <Book
      bookData={
        Object {
          "author": "apple lady2",
          "description": "who loves apples2",
          "picture": "http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
          "price": "502 usd",
          "title": "Apples2",
        }
      }
      key="1"
    />
    <Book
      bookData={
        Object {
          "author": "apple lady3",
          "description": "who loves apples3",
          "picture": "http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
          "price": "503 usd",
          "title": "Apples3",
        }
      }
      key="2"
    />
    <Book
      bookData={
        Object {
          "author": "apple lady4",
          "description": "who loves apples4",
          "picture": "http://books.google.com/books/content?id=FDEknQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
          "price": "504 usd",
          "title": "Apples4",
        }
      }
      key="3"
    />
  </ul>
</div>
`;
